HowToAddANewService.md

```diff
@@ -0,0 +1,157 @@
+
+
+(2) Setup
+---------
+
+I suppose you have [git](https://git-scm.com/) installed.
+If you do not, install it and learn about the [Github workflow](http://try.github.io/).
+
+1. [Fork][fork] this repository.
+2. Clone the fork  
+   `git clone git@github.com:YOURGITHUBUSERNAME/shields.git`
+3. `cd shields`
+4. Install npm  
+   `sudo apt-get install npm nodejs-legacy curl`
+5. Install all packages  
+   `npm install`
+6. Setup  
+   `make setup`
+7. Run the server  
+   `node server.js 8080`
+8. Visit the website to check the badges get loaded slowly:  
+   [http://[::1]:8080/try.html](http://[::1]/try.html)
+
+(3) Open an Issue
+-----------------
+
+Before you want to implement you service, you may want to [open an issue][openissue] and describe what you have in mind:
+- What is the badge for
+- Which API do you want to use
+
+You may additionally process to say what you want to work on.
+This infomration allows other humans to help and build on your work.
+
+(4) Implementing
+----------------
+
+Make sure, you can get the data somehow, best as JSON.
+There might be an API for your service delivering the data.
+
+If there is already a related badge, you may want to place your code next to it.
+
+Here you can see code for a badge at route for `/test/<first>/<second><ending>`.
+Lines with `// (1)` and alike are commented below.
+
+```
+// Test integration.
+camp.route(/^\/test\/([^\/]+)\/([^\/]+)\.(svg|png|gif|jpg|json)$/, // (1)
+cache(function(data, match, sendBadge, request) {
+  var first = match[1];     // (2)
+  var second = match[2];    // (2)
+  var format = match[3];    // (2)
+  var badgeData = getBadgeData('X' + first + 'X', data); // (3)
+  badgeData.text[1] = second;     // (4)
+  badgeData.colorscheme = 'blue'; // (4)
+  badgeData.colorB = '#008bb8';   // (4)
+  sendBadge(format, badgeData);   // (5)
+}));
+```
+
+Description of the code:
+
+1. This is the route specified as regular expression.
+2. These lines parse the route given in (1).
+3. This requests a default configuration for the badge.
+4. These lines customize the design of the badge.
+   Possible attributes are [specified][format].
+5. This line sends the response back to the browser.
+
+To try out this custom badge do the following:
+
+1. Copy and paste these lines into [server.js].
+   I did this at about line 5000.
+2. Quit the running server with `Control+C`.
+3. Start the server again.  
+   `node server.js 8080`
+4. Visit the badge at [http://[::1]:8080/test/asd/fgh.svg](http://[::1]:8080/test/asd/fgh.svg).
+
+### (4.1) Querying an API
+
+The test badge above is quite simple and static.
+You can change the behavior by asking your API.
+There are various examples on how to answer with a customized badge:
+
+- [docker automated builds](https://github.com/badges/shields/blob/bf373d11cd522835f198b50b4e1719027a0a2184/server.js#L5014)
+- [the famous travis badge](https://github.com/badges/shields/blob/bf373d11cd522835f198b50b4e1719027a0a2184/server.js#L431)
```

Comment:
I recognized that you created a new tag for such contributions. I would rather refer to that tag than the two example: https://github.com/badges/shields/pulls?utf8=%E2%9C%93&q=is%3Apr%20label%3Anew-badge%20